picard/util/pipe.py

```diff
@@ -137,6 +139,8 @@ def __init__(self, app_name: str, app_version: str, args: Optional[Iterable[str]
         if not isinstance(app_name, str) or not isinstance(app_version, str):
             raise PipeErrorInvalidAppData
 
+        self._identifier = identifier.replace(' ', '_') if identifier else 'main'
```

Comment:
@rdswift can `identifier` contain spaces the way it is build atm?

Reply:
No.  Good catch.  I had originally included the `.replace()` because I was first passing the whole config file path (which could contain spaces) but then decided to use the hash and forgot to remove the `.replace()`.

Reply:
I don't see any way to add spaces to the identifier but I think it is a good security feature that may prevent future errors.

Reply:
Or we can simply add a test case for pipe where we would assert that this variable contains no spaces, which gives better feedback for the developer

Reply:
I mean, the question is up to you @rdswift 

Could whitespaces in this string potentially break Picard?

If yes, I will add a proper test case. If not, I will remove the `.replace()` call :)

Reply:
I actually added it because I wasn't sure if a space would break the pipe name.  :stuck_out_tongue_winking_eye:   If that's not a concern, then there's likely no reason to keep it.